packages/engine-server/src/workspace/service.ts

```diff
@@ -1446,7 +1454,7 @@ export class WorkspaceService implements Disposable, IWorkspaceService {
       await Promise.all(
         _.map(workspaces, async (wsEntry, wsName) => {
           const wsPath = path.join(wsRoot, wsName);
-          if (!fs.existsSync(wsPath)) {
+          if (!(await fs.pathExists(wsPath))) {
```

Comment:
was there a reason this was converted from the sync version?

Reply:
It's already in an async function, so it makes sense to use the async version to me. Avoiding blocking for file operations makes the engine more efficient.